tests/e2e/testdata/scenarios/direct/llm_immutable_field/labels/_object01.yaml

```diff
@@ -0,0 +1,75 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: logging.cnrm.cloud.google.com/v1beta1
+kind: LoggingLogMetric
+metadata:
+  annotations:
+    cnrm.cloud.google.com/management-conflict-prevention-policy: none
+    cnrm.cloud.google.com/project-id: ${projectId}
+    cnrm.cloud.google.com/state-into-spec: absent
+  finalizers:
+  - cnrm.cloud.google.com/finalizer
+  - cnrm.cloud.google.com/deletion-defender
+  generation: 3
+  name: logginglogmetric-${uniqueId}
+  namespace: ${projectId}
+spec:
+  bucketOptions:
+    explicitBuckets:
+      bounds:
+      - 1.5
+      - 4.5
+  description: an updated description
+  filter: resource.type=gae_app AND severity<=ERROR
+  labelExtractors:
+    testkey: EXTRACT(jsonPayload.request)
+  metricDescriptor:
+    displayName: a concise name
+    labels:
+    - description: a label description
+      key: testkey
+      valueType: INT64
+    launchStage: EARLY_ACCESS
+    metadata:
+      ingestDelay: 1s
+      samplePeriod: 1s
+    metricKind: DELTA
+    valueType: DISTRIBUTION
+  projectRef:
+    external: projects/${projectId}
+  resourceID: logginglogmetric-${uniqueId}
+  valueExtractor: EXTRACT(jsonPayload.response)
+status:
+  conditions:
+  - lastTransitionTime: "1970-01-01T00:00:00Z"
+    message: |-
+      Update call failed: error applying desired state: googleapi: Error 400: Cannot change value type of label: testkey
+      Details:
+      [
+        {
+          "@type": "type.googleapis.com/google.rpc.DebugInfo",
+          "detail": "[ORIGINAL ERROR] generic::invalid_argument: Cannot change value type of label: testkey [google.rpc.error_details_ext] { message: \"Cannot change value type of label: testkey\" }"
+        }
+      ]
```

Comment:
I captured the golden files using DCL controller, the error messages are slightly different between DCL and direct. 
i.e. DCL: `Update call failed: error applying desired state: googleapi: Error 400: Cannot change value type of label: testkey`, direct: `Update call failed: logMetric update failed: googleapi: Error 400: Cannot change value type of label: testkey`

If we run the golden check with direct controller, the test will fail because of the diff in the error message. That is expected and validates that the direct controller behaves the same as DCL (with a different error message). I want to leave a note here for future reference in case anyone encounters issues with golden tests.